tower/src/retry/mod.rs

```diff
@@ -17,9 +17,20 @@ pin_project! {
     /// Configure retrying requests of "failed" responses.
     ///
     /// A [`Policy`] classifies what is a "failed" response.
+    ///
+    /// # Clone
+    ///
+    /// This middleware requires that the `Service` is `Clone` due to requirements
+    /// of `'static` futures. To easily add `Clone` to your service you can
+    /// use the `Buffer` middleware.
```

Comment:
some minor docs nits:
 - "due to requirements of `'static` futures" feels kind of vague to me --- will the reader understand what that means here?
 - in general, we tend not to use second-person ("you") language in API docs.

how about something like this:
```suggestion
    /// This middleware requires that the inner `Service` implements [`Clone`],
    /// because the `Service` must be stored in each [`ResponseFuture`] in
    /// order to retry the request in the event of a failure. If the inner
    /// `Service` type does not implement `Clone`, the [`Buffer`] middleware
    /// can be added to make any `Service` cloneable.
    ///
    /// [`Buffer`]: crate::buffer::Buffer
```